### DISCUSSION TITLE
How do I update every item on a GUI if one item change it status?

### DISCUSSION THREAD

**Discussion by sdxqw** [NichtStudioCode/InvUI#79] [Q&A]
title.

**Comment by NichtStudioCode** [comment]
https://docs.xen.cx/invui/items/

**Comment by sdxqw** [comment]
Not usefull, i have a voting system, and the voting acount is in the display name which it doesn't update

**Reply by NichtStudioCode** [reply]
I can only reiterate what's already explained in the docs. If you want to update an item, call notifyWindows on it.

**Reply by SorkoPiko** [reply]
that only updates the one item. invui is lacking a way to re-render all items, which is really annoying. i've had to resort to doing this:
```java
public class RefreshableSingleWindow extends AbstractSingleWindow {

    /**
     * Creates a new {@link RefreshableSingleWindow}.
     *
     * @param viewer    The player that views the window.
     * @param title     The title of the window.
     * @param gui       The gui of the window.
     * @param closeable Whether the window is closeable.
     */
    public RefreshableSingleWindow(Player viewer, ComponentWrapper title, AbstractGui gui, boolean closeable) {
        super(viewer, title, gui, InventoryUtils.createMatchingInventory(gui, ""), closeable);
    }

    public void refresh() {
        for (int i = 0; i < getGui().getSize(); i++) {
            redrawItem(i);
        }
    }
}
```

**Reply by NichtStudioCode** [reply]
Generally the idea there is that you don't actually want to update all items, but only a certain subset depending on what state changed, so you remember those item groups and then update them.
Refreshing on a window-level doesn't make much sense because it would create a desync between different windows that show the same gui, if they exist.
You may also want to take a look at the alpha versions of v2 if you're looking for utility functions for mass-updating items on a gui-level.

**Reply by SorkoPiko** [reply]
is v2 stable yet for big production servers?

**Reply by NichtStudioCode** [reply]
The only big issue I'm currently aware of is that there are no bukkit InventoryClickEvents fired for interactions with the player inventory while in a window. But you should probably evaluate this for yourself.
